build/go.mod

```diff
@@ -1,11 +1,55 @@
 module github.com/mattermost/mattermost-plugin-starter-template/build
```

Comment:
I wonder if we should abandon the nested `build/go.mod` and just rely on the top-level one? I don't think we need to manage versions independently, and Go will already only compile what it actually needs.

Reply:
@lieut-data Personally, I like the `build/go.mod` file being separate, because there are dependencies needed there that aren't needed for the source code of a given plugin. And although those dependencies are not included in the compiled plugin, having unrelated dependencies in the top-level `go.mod` file can add noise to what the given plugin is "all about".

There is also somewhat essentially dead code in the `build` folder, more specifically `build/sync`, which is using the dependency `github.com/go-git/go-git/v5`. If we were move the `build/go.mod` dependencies into `go.mod`, then I would also suggest to remove this `build/sync` folder to remove that dependency.

Saying this out loud is making me think we should probably remove that folder regardless of the `build/go.mod` decision. It was an attempt at having synchronization logic between plugin repos https://github.com/mattermost/mattermost-plugin-starter-template/pull/86, but it unfortunately didn't end up being used in our tooling. Not part of this PR, though wondering thoughts on removing this as a means of code cleanup.

Reply:
:+1: on removing `build/sync`. 0/5 on consolidating `go.mod` -- happy to defer to your thinking above :)